src/main/java/core/basesyntax/dao/FruitDaoImpl.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import java.util.Map;
+import java.util.Set;
+
+public class FruitDaoImpl implements FruitDao {
+    @Override
+    public void add(Fruit fruit, int amount) {
+        Storage.fruits.put(fruit, amount);
+    }
+
+    @Override
+    public Fruit getFruit(String fruitName) {
+        return Storage.fruits.keySet().stream()
+                .filter(integer -> integer.getFruitName().equals(fruitName))
+                .findFirst()
+                .get();
+    }
+
+    @Override
+    public int getAmount(String fruitName) {
+        return Storage.fruits.entrySet().stream()
+                .filter(n -> n.getKey().getFruitName().equals(fruitName))
+                .map(Map.Entry::getValue)
+                .findFirst()
+                .get();
```

Comment:
It would be easier to pass the Fruit by itself and use method get of HashMap

Reply:
either answer the comment or solve it

Reply:
sorry, I forgot about it